### PR TITLE
[bitnami/jenkins] Chart standardised

### PR DESCRIPTION
**Pull request by migruiz4**
**Description of the change**

This PR standardizes the bitnami/jenkins chart to be in line with the rest of the catalog.

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by migruiz4** [comment]
Fixes #9027

File: bitnami/jenkins/templates/deployment.yaml [hunk 1]
```diff
@@ -139,6 +146,17 @@ spec:
             - name: https
               containerPort: {{ .Values.containerPorts.https }}
               protocol: TCP
+          {{- if not .Values.diagnosticMode.enabled }}
+          {{- if .Values.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
```
Comment (juan131) [inline bitnami/jenkins/templates/deployment.yaml:152]: Why would you exclude the hook when enabling diagnostic mode?
```suggestion
          {{- if .Values.lifecycleHooks }}
          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.lifecycleHooks "context" $) | nindent 12 }}
          {{- end }}
          {{- if not .Values.diagnosticMode.enabled }}
```

File: bitnami/jenkins/templates/deployment.yaml [hunk 2]
```diff
@@ -220,7 +239,7 @@ spec:
         - name: jenkins-data
           {{- if .Values.persistence.enabled }}
           persistentVolumeClaim:
-            claimName: {{ include "common.names.fullname" . }}
+            claimName: {{ default (include "common.names.fullname" .) .Values.persistence.existingClaim }}
```
Comment (juan131) [inline bitnami/jenkins/templates/deployment.yaml:242]: Support templates in `.Values.persistence.existingClaim`:
```suggestion
            claimName: {{ default (include "common.names.fullname" .) (tpl .Values.persistence.existingClaim $) }}
```

File: bitnami/jenkins/templates/pvc.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.persistence.enabled -}}
+{{- if and .Values.persistence.enabled (not .Values.persistence.existingClaim) -}}
```
Comment (juan131) [inline bitnami/jenkins/templates/pvc.yaml:1]: Am I missing `persistence.selector`?

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the PR @migruiz4 

Please remove the block below in the NOTES.txt:
```
{{- $passwordValidationErrors := list -}}
{{- $secretName := include "common.names.fullname" . -}}
{{- $requiredJenkinsPassword := dict "valueKey" "jenkinsPassword" "secret" $secretName "field" "jenkins-password" "context" $ -}}
{{- $requiredJenkinsPasswordError := include "common.validations.values.single.empty" $requiredJenkinsPassword -}}
{{- $passwordValidationErrors = append $passwordValidationErrors $requiredJenkinsPasswordError -}}
{{- include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordValidationErrors "context" $) -}}
```
The current secret uses "lookup" functionalities so these validations aren't required., see: https://github.com/bitnami/charts/blob/master/bitnami/jenkins/templates/secrets.yaml#L15

File: bitnami/jenkins/templates/pvc.yaml (hunk 3 above)
Comment (migruiz4) [reply bitnami/jenkins/templates/pvc.yaml:1]: Nice catch! Implemented.

**Comment by migruiz4** [comment]
I applied all the suggested changes

**Review by juan131 — DISMISSED** [review]
lgtm

**Pull request merged**